domain/src/main/java/org/oppia/android/domain/exploration/lightweightcheckpointing/ExplorationStorageModule.kt

```diff
@@ -0,0 +1,26 @@
+package org.oppia.android.domain.exploration.lightweightcheckpointing
+
+import dagger.Module
+import dagger.Provides
+import javax.inject.Qualifier
+
+@Qualifier
+annotation class ExplorationStorageDatabaseSize
+
+/** Provider to return any constants required during the storage of exploration checkpoints. */
+@Module
+class ExplorationStorageModule {
+
+  /**
+   * Provides the size allocated to exploration checkpoint database.
+   *
+   * The current [ExplorationStorageDatabaseSize] is set to 2097152 Bytes that is equal to 2MB
+   * per profile.
+   *
+   * Taking 20 KB per checkpoint, it is expected to store about 100 checkpoints for every profile
+   * before the database exceeds the allocated limit.
+   */
+  @Provides
+  @ExplorationStorageDatabaseSize
+  fun provideExplorationStorageDatabaseSize(): Int = 2097152
```

Comment:
I was thinking to keep this constant in the build config or maybe a config file since it's quite easy to change this constant from here. @BenHenning @rt4914 Your thoughts ?

Reply:
We can keep it anywhere but is possible to add a test case where can check this value ? If yes, then this value can be kept anywhere else your idea sounds good.

Reply:
@MaskedCarrot let's move this to build config

Reply:
Using a dagger constant as per @vinitamurthi suggestion